src/platform/silabs/efr32/KeyValueStoreManagerImpl.cpp

```diff
@@ -67,25 +70,63 @@ bool KeyValueStoreManagerImpl::IsValidKvsNvm3Key(uint32_t nvm3Key) const
     return ((SilabsConfig::kConfigKey_KvsFirstKeySlot <= nvm3Key) && (nvm3Key <= SilabsConfig::kConfigKey_KvsLastKeySlot));
 }
 
-CHIP_ERROR KeyValueStoreManagerImpl::MapKvsKeyToNvm3(const char * key, uint32_t & nvm3Key, bool isSlotNeeded) const
+uint16_t KeyValueStoreManagerImpl::hashKvsKeyString(const char * key) const
+{
+    uint8_t hash256[Crypto::kSHA256_Hash_Length] = { 0 };
+    Crypto::Hash_SHA256(reinterpret_cast<const uint8_t *>(key), strlen(key), hash256);
+
+    uint16_t hash16, i = 0;
+
+    while (!hash16 && (i < (Crypto::kSHA256_Hash_Length - 1)))
+    {
+        hash16 = (hash256[i] | (hash256[i + 1] << 8));
+        i++;
+    }
+    return hash16;
+}
+
+CHIP_ERROR KeyValueStoreManagerImpl::MapKvsKeyToNvm3(const char * key, uint16_t hash, uint32_t & nvm3Key, bool isSlotNeeded) const
```

Comment:
Adding a check for the key would save some processing time in the case of a `nullptr`

Reply:
This function MapKvsToNvm3 is only called by one of the main functions (get, put, delete) and those functions already have a check for a nullptr key. I can add it if you want but I believe it is redundant and adds a extra call that will never assert.

Reply:
In that case we can do without